fe1-web/store/reducers/SocialReducer.ts

```diff
@@ -24,13 +28,33 @@ interface SocialLaoReducerState {
 const initialState: SocialLaoReducerState = {
   byLaoId: {
     myLaoId: {
-      allChirps: [],
+      allIdsInOrder: [],
+      byId: {},
+      byUser: {},
     },
   },
 };
 
 const socialReducerPath = 'social';
 
+// helper function to find where to insert the new chirp in ascending order
+function findInsertIdx(
+  array: string[], byId: Record<string, ChirpState>, element: number,
```

Comment:
Good job on the efficient implementation.
I might have been lazier :laughing:

Reply:
🤩 I would love to be lazier, can I have a hint of that, please?

Reply:
Lazier as in: inefficient. I might have done a linear search and optimized later "if needed" 😄 
But really, your implementation is much better 👍